### ISSUE TITLE
vMware suggests I ask here so here it goes

### ISSUE THREAD

**Issue by anotherPassword** [jakkulabs/PowervRA#241]
I have a fresh vra 8.1 installation. As the documentation states, it includes powerVRA. After some investigation, it seems that the included version of powervra is 3.6.0. I believe I need to be at version 4.1.1 to be compatible with the 8.1 API. Using and ABX or a scriptable task, I am able to run update-module and install-module. Both work. But they seem to afffect '/root/.local/share/powershell/Modules/PowervRA/4.1.1''. When I attempt to use the module, it is still at 3.6 version. For some reason, vMware keeps telling me they don't support powerVRA, which I know is true, but how to I powerVRA up to latest version on the fresh 8.1 installation? I am thinking since it is container related, there is some base container I have to access to update this. Thank you for any help.
ds

**Comment by mpuckett159** [comment]
I've never tried this but you might be able to try following this blog from VMware regarding how to manually inject script dependencies into an action.

https://blogs.vmware.com/management/2020/07/how-to-abx-vro-powershell-additional-modules.html

Alternatively you could try running and Update-Module PowerVRA at the beginning of every action where you want to run the most recent version, assuming your vRA instance has access to the internet.

Basically, what sounds like it happening is you are updating the spawned container that executes the Action, but then trying to run another, different action afterwards. Each ABX Action runs independently of every other action. It is a Function as a Service model similar to Lambda and Azure Functions, and so updates you make in an action are not carried over between actions, thus you need to update the module in every run.

**Comment by anotherPassword** [comment]
Thank you Marcus. I am for sure updating the spawned container, I figured that eventually after looking at what it was doing. I do agree I should be able to update or install module on ABX run and it should work, but that to me do e not address the root issue. What I think I should be able to do is access whatever it is that the container spawns from, which has the 3.6.0 version installed, access it, uninstall 3.6.0, and install 4.1.1. Then each time a container spawns, it will have the right version.
The thing that is sticking me is that I can not get vMware to help on how to update that initial instance of what it is the container spawns from.

**Comment by mpuckett159** [comment]
Yeah they aren't gonna help you with that and I wouldn't recommend doing that. The supported and recommended way to import modules not already included in the container is outlined in the blog post above.

**Comment by anotherPassword** [comment]
That's helpful, I will start exploring the other option more in depth. I was just floored no-one there checked the version of powerVRA before they included it. Thanks again.
